crates/wasi-common/src/virtfs.rs

```diff
@@ -78,7 +78,7 @@ impl FileContents for VecFileContents {
     fn preadv(&self, iovs: &mut [io::IoSliceMut], offset: types::Filesize) -> Result<usize> {
         let mut read_total = 0usize;
         for iov in iovs.iter_mut() {
-            let read = self.pread(iov, offset)?;
+            let read = self.pread(iov, offset + read_total as u64)?;
```

Comment:
Since there is no auto conversion from `usize` to `u64`, I added a temporary variable with type annotation and used `try_into`, but I suppose there is a simpler solution.

Btw, while doing that I also noticed `fd_pwrite` has the same issue, so I included a fix for that as well.